src/gc/gc.cpp

```diff
@@ -15882,14 +15882,42 @@ start_no_gc_region_status gc_heap::prepare_for_no_gc_region (uint64_t total_size
     save_data_for_no_gc();
     settings.pause_mode = pause_no_gc;
     current_no_gc_region_info.start_status = start_no_gc_success;
-    
+
     size_t allocation_no_gc_loh = 0;
     size_t allocation_no_gc_soh = 0;
     size_t size_per_heap = 0;
 
+    // requested sizes of 0 make no sense.
+    if (total_size == 0)
+    {
+        status = start_no_gc_too_large;
```

Comment:
This seems weird to me. Why are we reusing the 'too large' status for when size is 0? Is it a point-in-time thing that will be addressed soon? This is what happens in `StartNoGCRegionWorker` when this 'too large' case occurs:
```
if (status == StartNoGCRegionStatus.AmountTooLarge)
                throw new ArgumentOutOfRangeException(nameof(totalSize),
                    "totalSize is too large. For more information about setting the maximum size, see \"Latency Modes\" in http://go.microsoft.com/fwlink/?LinkId=522706");
```

Reply:
yep... Ideally we'd surface this as a new error all the way up to managed code. i'm not sure what the compat implications of this are, which is why I did this. If I have the freedom to do this across our various runtimes, I'd be happy to do so - I'm just not sure if I do.

Reply:
I suppose as long as `StartNoGCRegionStatus` is private, there's no problem, and this is technically already a breaking change already...